2101 User Guide: Event and Enrolment Commands (#288)
AY2324S1-CS2103T-F11-1/tp: docs/2101UserGuide.md

`editEvent 1 l/MBS d/2023-10-20`

<hr class="command-separator">

#### Delete an Event: `deleteEvent`

Are the number of events piling up? To clear the clutter, you can delete any unwanted event from CCACommander with `deleteEvent`!

Here is how you can use it:<br>
`deleteEvent EVENT_INDEX`

Here is an example command where we delete the event at **index** 1:<br>
`deleteEvent 1`

<div markdown="block" class="alert alert-info">:information_source: The `EVENT_INDEX` parameter refers to the index number shown in the **currently displayed** event list.
</div>

<hr class="feature-class-separator">

### Enrolment Commands

Enrolments are a way for you to manage your members’ involvement in your events. The enrolment commands below are the bread-and-butter for your participation tracking needs:

#### Enrol a Member to an Event: `enrol`

Now that you have added all your members and all your events, keeping track of your members’ attendance to any event is as simple as using the `enrol` command!

Here’s how you can use it:<br>
`enrol m/MEMBER_INDEX e/EVENT_INDEX [h/NUMBER_OF_HOURS] [r/REMARK]`

> Here is an example command where we enrol the member at **member index** 1 to the event at **event index** 1, where our member
> has contributed 2 **hours**, and we want to make a "Role: Photographer" **remark** to reflect their role as a photographer:<br>

Reviewer: Do you think this will be better:
```suggestion
Here is an example command where we enrol the member at **member index** 1 to the event at **event index** 1, where our member
has contributed 2 **hours**, and we want to leave a **remark** "Role: Photographer" to reflect their role as a photographer:<br>
```

Author: Done!

`enrol m/1 e/1 h/2 r/Role: Photographer`

<hr class="command-separator">

#### Edit an enrolment: `editEnrolment`
Accidentally mixed up two members, and gave them the wrong number of hours or wrong remarks for the event they are involved in?
You can fix this with ease by using the `editEnrolment` command!

Here’s how you can use it:<br>
`editEnrolment m/MEMBER_INDEX e/EVENT_INDEX [h/NUMBER_OF_HOURS] [r/REMARK]`

<div markdown="block" class="alert alert-info">
:information_source: At least one field must be provided.
</div>

Here is an example command where we edit the enrolment of the member at **member index** 1 to the event at **event index** 2,
changing the number of **hours** to 5 and the **remark** to "Role: Exco" to reflect the member’s role as an exco member:<br>
`editEnroment m/1 e/2 h/5 r/Role: Exco`

<hr class="command-separator">

#### Unenrol a Member from an Event: `unenrol`
If your member is no longer involved in a particular event, you can easily remove them from the event by using the `unenrol` command!

Here’s how you can use it:<br>
`unenrol m/MEMBER_INDEX e/EVENT_INDEX`

> Here is an example command where we unenrol the **member** at index 1 from the **event** at index 1:<br>

Reviewer: Should we update this to follow the format above?
```suggestion
Here is an example command where we unenrol the member at **member index** 1 from the event at **event index** 1:<br>
```

<hr class="feature-class-separator">

### Event Commands

As a CCA head, keeping track of event details is a major responsibility. CCACommander allows you to do so with the event-related commands below:

#### Create an Event : `createEvent`

Creating a new event is no longer a hassle as it can be done with just one simple command - `createEvent`!

Here is how you can use it:<br>
`createEvent n/EVENT_NAME l/LOCATION d/DATE [t/TAG]...`

Here is an example command where we create an event **named** "Party" held at the **location** "Raffles Hall" on the
**date** 2023-09-16, with **tag** "Dinner" and **tag** "Sem1":<br>
`createEvent n/Party l/Raffles Hall d/2023-09-16 t/Dinner t/Sem1`

<figure>
    <img src="images/CreateEventPostCommand.png"
         alt="createEvent Post Command">
    <figcaption>After executing the `createEvent` command</figcaption>
</figure>

<hr class="command-separator">

> #### Edits an Event : `editEvent`

Reviewer 2: I think no s here to match the other commands
```suggestion
#### Edit an Event : `editEvent`
```

Reviewer 3: agreed

If you forgot to note any important details, or realise that you made a mistake in any field of the event created,
the `editEvent` command is here to help.

Here is how you can use it:<br>

> `editEvent EVENT_INDEX [n/EVENT_NAME] [l/LOCATION] [d/DATE] [t/TAG]...`

Reviewer: I think we can add an info card here, telling the user that at least 1 field is required

> Here is an example command where we edit an **event** at index 1, changing the **location** to be "MBS" and the **date** to be "2023-10-20".<br>

Reviewer 2: I think index should be bolded instead of event, cause the parameteres are bolded so it's 
```suggestion
Here is an example command where we edit an event at **index** 1, changing the **location** to be "MBS" and the **date** to be "2023-10-20".<br>
```

`editEvent 1 l/MBS d/2023-10-20`

<hr class="command-separator">

#### Delete an Event: `deleteEvent`

Are the number of events piling up? To clear the clutter, you can delete any unwanted event from CCACommander with `deleteEvent`!

Here is how you can use it:<br>
`deleteEvent EVENT_INDEX`

> Here is an example command where we delete the event at index 1:<br>

Reviewer 3: Do you want to consider bolding here too?

`deleteEvent 1`

> <div markdown="block" class="alert alert-primary">:bulb: The `EVENT_INDEX` parameter refers to the index number shown in the **currently displayed** event list.

Reviewer: I think we should change this to an info block instead, to match the comment you left for the `editMember` command.

For reference:
https://github.com/AY2324S1-CS2103T-F11-1/tp/pull/287#discussion_r1390405538

</div>

<hr class="feature-class-separator">

### Enrolment Commands

Enrolments are a way for you to manage your members’ involvement in your events. The enrolment commands below are the bread-and-butter for your participation tracking needs:

#### Enrol a Member to an Event: `enrol`

Now that you have added all your members and all your events, keeping track of your members’ attendance to any event is as simple as using the `enrol` command!

Here’s how you can use it:<br>
`enrol m/MEMBER_INDEX e/EVENT_INDEX [h/NUMBER_OF_HOURS] [r/REMARK]`

> Here is an example command where we want to enrol the **member** at index 1 to the **event** at index 1, where our member

Reviewer 2: ```suggestion
Here is an example command where we want to enrol the **member** at **index** 1 to the **event** at **index** 1, where our member
```

> has contributed 2 **hours**, and we want to make a "Role: Photographer" **remark** to reflect their role as a photographer:<br>

Reviewer 2: should we change this for brevity or do you think the original is more user-friendly
```suggestion
has contributed 2 **hours**, and we want to make a "Role: Photographer" **remark**:<br>
```

Reviewer: I agree with @ph-nathan, we can remove that part

Author: I feel that leaving it will seem more conversational rather than so surgical, what do you think?

Reviewer 3: I agree with leaving that line in

`enrol m/1 e/1 h/2 r/Role: Photographer`

<hr class="command-separator">

#### Edit an enrolment: `editEnrolment`
Accidentally mixed up two members, and gave them the wrong number of hours or wrong remarks for the event they are involved in?

> You can fix this with no problem by using the `editEnrolment` command!

Reviewer 3: ```suggestion
You can fix this with ease by using the `editEnrolment` command!
```
Not too sure which sounds better, for some reason `with no problem` took a while for me

Here’s how you can use it:<br>
`editEnrolment m/MEMBER_INDEX e/EVENT_INDEX [h/NUMBER_OF_HOURS] [r/REMARK]`

> Here is an example command where we want to edit the enrolment of the **member** at index 1 to the **event** at index 2,

Reviewer 2: ```suggestion
Here is an example command where we want to edit the enrolment of the **member** at **index** 1 to the **event** at **index** 2,
```

> Here is an example command where we want to edit the enrolment of the **member** at index 1 to the **event** at index 2,

Reviewer: I think we can keep it more concise by shortening it to:
```suggestion
Here is an example command where we edit the enrolment of the member at **index** 1 to the event at **index** 2,
```

Reviewer 3: Agreed with removing the `want to`

> changing the number of **hours** to 5 and the **remark** to "Role: Exco" to reflect the member’s role as an exco member:<br>

Reviewer 2: should we change this for brevity or do you think the original is more user-friendly
```suggestion
changing the number of **hours** to 5 and the **remark** to "Role: Exco":<br>
```

`editEnroment m/1 e/2 h/5 r/Role: Exco`

<hr class="command-separator">

#### Unenrol a Member from an Event: `unenrol`
If your member is no longer involved in a particular event, you can easily remove them from the event by using the `unenrol` command!

Here’s how you can use it:<br>
`unenrol m/MEMBER_INDEX e/EVENT_INDEX`

> Here is an example command where we want to unenrol the **member** at index 1 from the **event** at index 1:<br>

Reviewer 3: ```suggestion
Here is an example command where we unenrol the **member** at index 1 from the **event** at index 1:<br>
```
Similar suggestion above to make it more concise

`unenrol m/1 e/1`

> <div markdown="block" class="alert alert-primary">:bulb: The `MEMBER_INDEX`/`EVENT_INDEX` parameter refers to the index number shown in the **currently displayed** member/event list.

Reviewer 3: I think we can change this to ℹ️  as mentioned by andre as well
